cloudslang-content-maven-compiler/src/main/java/io/cloudslang/maven/compiler/CloudSlangMavenCompiler.java

```diff
@@ -0,0 +1,148 @@
+        if(sourceFiles.length > 0) {
+            System.out.println("Compiling " + sourceFiles.length + " " + "source file" + (sourceFiles.length == 1?"":"s"));
+            for (String sourceFile : sourceFiles) {
+                compilerMessage.addAll(compileFile(sourceFile));
+            }
+
+            if (compilerMessage.size() > 0){
+                compilerResult.setCompilerMessages(compilerMessage);
+                compilerResult.setSuccess(false);
+            }
+        }
+
+        return compilerResult;
+    }
+
+    private List<CompilerMessage> compileFile(String sourceFile){
+        ExecutableModellingResult executableModellingResult ;
+        List<CompilerMessage> compilerMessage = new ArrayList();
+
+
+        try {
+            SlangSource slangSource = SlangSource.fromFile(new File(sourceFile));
+            executableModellingResult = slangCompiler.preCompileSource(slangSource);
+            if(executableModellingResult.getErrors() != null && !executableModellingResult.getErrors().isEmpty()){
+                for (RuntimeException runtimeException : executableModellingResult.getErrors()) {
+                    compilerMessage.add(new CompilerMessage(sourceFile + ": " + runtimeException.getMessage(), CompilerMessage.Kind.ERROR)) ;
+                }
+            }
+        } catch (Exception e) {
+            compilerMessage.add(new CompilerMessage(sourceFile + ": " + e.getMessage(), CompilerMessage.Kind.ERROR)) ;
+        }
+
+        return compilerMessage;
+    }
+
+    public String[] createCommandLine(CompilerConfiguration config) throws CompilerException {
+        return null ;
+    }
+
+    protected static String[] getSourceFiles( CompilerConfiguration config ){
+        Set sources = new HashSet();
+
+        for(Iterator it = config.getSourceLocations().iterator(); it.hasNext();){
+            String sourceLocation = (String) it.next();
+            sources.addAll(getSourceFilesForSourceRoot(config, sourceLocation ));
+        }
+
+        String[] result;
+        if ( sources.isEmpty() ){
+            result = new String[0];
+        } else {
+            result = (String[]) sources.toArray( new String[sources.size()] );
+        }
+
+        return result;
+    }
+
+    // we need to override this as it is hard coded java file extensions
+    protected static Set getSourceFilesForSourceRoot(CompilerConfiguration config, String sourceLocation) {
+        Path path = Paths.get(sourceLocation);
+        if(!Files.exists(path)) return Collections.EMPTY_SET;
+
+        DirectoryScanner scanner = new DirectoryScanner();
+        scanner.setBasedir(sourceLocation);
+        Set includes = config.getIncludes();
+        if(includes != null && !includes.isEmpty()) {
+            String[] excludes = (String[])(includes.toArray(new String[includes.size()]));
+            scanner.setIncludes(excludes);
+        } else {
+            scanner.setIncludes(new String[]{"**/*.yaml","**/*.sl","**/*.yml"});
```

Comment:
okya, but do we want to compile all yaml files? IMHO only the ones that contain .sl